### PR TITLE
feat: capture test duration and failure details in build scans

### PR DESCRIPTION
**Pull request by tacascer**
## Summary
- Capture per-test duration (computed from frame timestamp deltas between TestCase/TestResult events) across the full stack
- Capture `failure_id` from wire 284 (previously read-and-discarded) using `FailureId` newtype for future ExceptionTree correlation
- Add scan-level test summary stats (passed/failed/skipped counts + total duration) as a GraphQL field
- Add Duration column, expandable failure detail rows, and summary badges to the Angular frontend

## Changes
- **Parser**: `test_result.rs` captures `failure_id`, `assembly.rs` computes `duration_ms` from frame timestamp deltas
- **Server**: New DB migration (008), `TestSummary` domain model, `test_summary()` aggregate query, GraphQL `durationMs`/`failureMessage`/`failureStacktrace` fields + `TestSummary` type
- **Frontend**: Duration column, clickable failed test rows with expandable stack trace panel, summary stat badges

## Follow-up (Phase 2)
- ExceptionTree wire format reverse-engineering to populate `failure_message`/`failure_stacktrace` columns (infrastructure is ready)

## Test plan
- [x] All 9 Rust tests pass (`bazelisk test //build-scan/...`)
- [x] Angular tests pass with updated assertions for new Duration column and summary stats
- [x] Build succeeds for all targets (`bazelisk build //build-scan/... //angular/...`)
- [ ] E2E: dogfooding with intentionally failing Gradle test shows duration + summary in UI

### CODE REVIEW

File: build-scan/server/graphql/src/lib.rs [hunk 1]
```diff
@@ -380,6 +389,45 @@ impl Test {
     fn outcome(&self) -> Option<String> {
         self.test.outcome.map(|o| o.to_string())
     }
+
+    fn duration_ms(&self) -> Option<i32> {
+        self.test.duration_ms.map(|d| d as i32)
```
[the remark this answers is not in the document]
Comment (tacascer) [reply build-scan/server/graphql/src/lib.rs]: Fixed in d83a1a3. Changed to i32::try_from with FieldResult error to avoid silent truncation.

File: build-scan/server/graphql/src/lib.rs [hunk 2]
```diff
@@ -380,6 +389,45 @@ impl Test {
...
+    fn failed(&self) -> i32 {
+        self.summary.failed as i32
+    }
+
+    fn skipped(&self) -> i32 {
+        self.summary.skipped as i32
+    }
+
+    fn total_duration_ms(&self) -> Option<i32> {
+        self.summary.total_duration_ms.map(|d| d as i32)
```
[the remark this answers is not in the document]
Comment (tacascer) [reply build-scan/server/graphql/src/lib.rs]: Fixed in d83a1a3. Same i32::try_from approach applied to total_duration_ms.

File: angular/projects/build-scan-web/src/app/scans/tests-table/tests-table.component.ts [hunk 3]
```diff
@@ -7,19 +12,42 @@ import { Component, ChangeDetectionStrategy, input } from "@angular/core";
...
+              <th>Duration</th>
               <th>Executor</th>
             </tr>
           </thead>
           <tbody>
             @for (edge of testEdges(); track edge.node.id) {
-              <tr>
+              <tr
+                [class.cursor-pointer]="edge.node.outcome === 'Failed' && edge.node.failureMessage"
+                (click)="toggleExpanded(edge.node.id)"
```
[the remark this answers is not in the document]
Comment (tacascer) [reply angular/projects/build-scan-web/src/app/scans/tests-table/tests-table.component.ts]: Fixed in d83a1a3. Click handler now only triggers for failed tests with failure details.

[the remark this answers is not in the document]
Comment (tacascer) [reply angular/projects/build-scan-web/src/app/scans/tests-table/tests-table.component.ts]: Fixed in d83a1a3. Added tabindex, aria-expanded, and keydown.enter/keydown.space handlers for keyboard accessibility.

File: build-scan/lib/src/assembly.rs [hunk 4]
```diff
@@ -598,9 +602,14 @@ pub fn assemble(events: Vec<(FramedEvent, DecodedEvent)>) -> BuildScanPayload {
...
-                .filter_map(|(tc, outcome)| {
+                .filter_map(|((tc, case_timestamp), result)| {
                     tc.map(|mut test_case| {
-                        test_case.outcome = outcome;
+                        if let Some((outcome, result_timestamp, failure_id)) = result {
+                            test_case.outcome = Some(outcome);
+                            let duration = result_timestamp - case_timestamp;
+                            test_case.duration_ms = if duration >= 0 { Some(duration) } else { None };
+                            test_case.failure_id = failure_id;
+                        }
```
[the remark this answers is not in the document]
Comment (tacascer) [reply build-scan/lib/src/assembly.rs:612]: Fixed in d83a1a3. Added assertions for duration_ms (timestamp delta computation) and failure_id propagation in the existing test, plus a new test for the negative-duration edge case.

File: angular/projects/build-scan-web/src/app/scans/scan-detail.component.ts [hunk 5]
```diff
@@ -73,6 +73,12 @@ const GET_BUILD_SCAN = gql`
           endCursor
         }
       }
+      testSummary {
+        passed
+        failed
+        skipped
+        totalDurationMs
+      }
```
[the remark this answers is not in the document]
Comment (tacascer) [reply angular/projects/build-scan-web/src/app/scans/scan-detail.component.ts]: Fixed in d83a1a3. testSummary is now guarded behind @include(if: $includeTests) so task-only pagination doesn't recompute test aggregates.
